docs: add design pattern for date entry (#1179)
pluralsight/pando: website/docs/learn/learn-pando/design-patterns/entering-dates.mdx

Reviewer: Overall a good start. Just remember to follow the template down to a T as much as possible. I've tried to update as much as I can.

---
tags: [Dates, DatePicker, Field]
title: 'Entering Dates'
---

In order to provide an accessible user experience, you should design and develop date picking experiences to use an Input rather than a Date Picker.

> ## Entering Dates

Reviewer: ```suggestion
```
This is a redundant title.

> ### Calendar Picker vs. Input

Reviewer: ```suggestion
## Calendar Picker vs. Input
```

> **Calendar Pickers** are most useful for scheduling applications, where the user is selecting an unknown date with the aid of calendar context. They primarily benefit sighted users using a pointing device or touchscreen and are not fully accessible.

Reviewer: ```suggestion
### Date Pickers

Date Pickers are most useful for scheduling applications, where the user is selecting an unknown date with the aid of calendar context. They primarily benefit sighted users using a pointing device or touchscreen and most importantly **are not fully accessible**.

Despite what many articles claim, Date Pickers cannot be made accessible either. To learn more about why you should avoid using a Date Picker we highly recommend you read the article:

[Maybe You Don't Need a Date Picker](https://adrianroselli.com/2019/07/maybe-you-dont-need-a-date-picker.html).
```

> The **Input** is accessible by default and is actually easier use in most cases, particularly when entering a _known_ date. However, it's worth noting that input with `type="date"` has known accessibility issues, so we use the `text` type instead.

Reviewer: ```suggestion
### Text Input

The text-based Input is accessible by default and is easier use in most cases where the user needs to enter a _known_ date which applies to most applications and what we recommend using in Pando.

:::danger date Inputs are not accessible

Please note we are specifically using the `type="text"` input element and **not** the `type="date"`. This is because the date version of an input is (shockingly) **not accessible**.

:::

```

> The **Input** is accessible by default and is actually easier use in most cases, particularly when entering a _known_ date. However, it's worth noting that input with `type="date"` has known accessibility issues, so we use the `text` type instead.

Reviewer: ```suggestion
### Text Input

The text-based Input is accessible by default and is easier use in most cases where the user needs to enter a _known_ date which applies to most applications and what we recommend using in Pando.

:::danger date Inputs are not accessible

Please note we are specifically using the `type="text"` input element and **not** the `type="date"`. This is because the date version of an input is (shockingly) **not accessible**.

:::
```

> If you're interested in a deeper exploration, check out [Maybe You Don't Need a Date Picker](https://adrianroselli.com/2019/07/maybe-you-dont-need-a-date-picker.html).

Reviewer: ```suggestion
```

> ### Text Input for Date Entry

Reviewer: ```suggestion
## Text Input for Date Entry
```

> For entering a date, we use a plain text input combined with the [useAutoFormatDate](../../../reference/react-hooks/useAutoFormatDate.mdx) hook to automatically format the input according to the selected pattern. The text input gives us a solid, accessible baseline, while the hook reduces the cognitive load on the user with regards to formatting and validation.

Reviewer: ```suggestion
In Pando, we recommend using the text-based Input combined with the [useAutoFormatdate](../../../reference/react-hooks/useAutoFormatDate.mdx) hook for displaying a way to entering a date.
```

> To use the hook, apply the returned props to a regular text input, and add messaging to communicate the expected format.

Reviewer: ```suggestion
In this example, we are creating a DateInput component that can be used anywhere in your application.
```

```jsx live showLineNumbers

> function DateInput(props) {

Reviewer: We don't want to use props here because It alters the Label as you can see in the example. Likewise, the help text you are using is too repetitive to the placeholder and should instead be a sentence like in our Figma UI kit examples. 

Something like: `Enter a date in the format of MM/DD/YY.` would be a better option and more human friendly because it's meeting our new brand identity of being holistic vs. mechanic (i.e. no life - just data).

Author: Ah, ok. Hadn't thought about that for the props.  I started with an example from somewhere else and just modified it a bit.  I'll take care of that and the helper text, which I thought I had updated, but must have missed it.
